### ISSUE TITLE
sw_kp bounds not updating

### ISSUE THREAD

**Issue by jklenzing** [rstoneback/pysat#198]
Discovered this while running things in ipython to track down the source of #195.  Confirmed that the data file is in the directory, but the routine that mocks up the the daily data files for Feb 2019 does not generate values past 2019-02-01.

Codes to duplicate error:
```
import pysat
kp = pysat.Instrument("sw", "kp", "")     
start = pysat.datetime(2019, 1, 1)
stop = pysat.datetime(2019, 2, 28)
kp.download(start, stop)
```
Output:
```
Downloading data to:  /Users/jklenzing/data/pysat/sw/kp/
Downloading file for 01/01/2019
Downloading file for 01/02/2019
Downloading file for 01/03/2019
Downloading file for 01/04/2019
Downloading file for 01/05/2019
Downloading file for 01/06/2019
Downloading file for 01/07/2019
Downloading file for 01/08/2019
Downloading file for 01/09/2019
Downloading file for 01/10/2019
Downloading file for 01/11/2019
Downloading file for 01/12/2019
Downloading file for 01/13/2019
Downloading file for 01/14/2019
Downloading file for 01/15/2019
Downloading file for 01/16/2019
Downloading file for 01/17/2019
Downloading file for 01/18/2019
Downloading file for 01/19/2019
Downloading file for 01/20/2019
Downloading file for 01/21/2019
Downloading file for 01/22/2019
Downloading file for 01/23/2019
Downloading file for 01/24/2019
Downloading file for 01/25/2019
Downloading file for 01/26/2019
Downloading file for 01/27/2019
Downloading file for 01/28/2019
Downloading file for 01/29/2019
Downloading file for 01/30/2019
Downloading file for 01/31/2019
Downloading file for 02/01/2019
Downloading file for 02/02/2019
Downloading file for 02/03/2019
Downloading file for 02/04/2019
Downloading file for 02/05/2019
Downloading file for 02/06/2019
Downloading file for 02/07/2019
Downloading file for 02/08/2019
Downloading file for 02/09/2019
Downloading file for 02/10/2019
Downloading file for 02/11/2019
Downloading file for 02/12/2019
Downloading file for 02/13/2019
Downloading file for 02/14/2019
Downloading file for 02/15/2019
Downloading file for 02/16/2019
Downloading file for 02/17/2019
Downloading file for 02/18/2019
Downloading file for 02/19/2019
Downloading file for 02/20/2019
Downloading file for 02/21/2019
Downloading file for 02/22/2019
Downloading file for 02/23/2019
Downloading file for 02/24/2019
Downloading file for 02/25/2019
Downloading file for 02/26/2019
Downloading file for 02/27/2019
Downloading file for 02/28/2019
Updating pysat file list
pysat is searching for sw kp files.
Found 32 of them.
Updating instrument object bounds.
```
Checking for available files:
```
kp.files.files
```
Output:
```
2019-01-01    kp1901.tab_2019-01-01
2019-01-02    kp1901.tab_2019-01-02
2019-01-03    kp1901.tab_2019-01-03
2019-01-04    kp1901.tab_2019-01-04
2019-01-05    kp1901.tab_2019-01-05
2019-01-06    kp1901.tab_2019-01-06
2019-01-07    kp1901.tab_2019-01-07
2019-01-08    kp1901.tab_2019-01-08
2019-01-09    kp1901.tab_2019-01-09
2019-01-10    kp1901.tab_2019-01-10
2019-01-11    kp1901.tab_2019-01-11
2019-01-12    kp1901.tab_2019-01-12
2019-01-13    kp1901.tab_2019-01-13
2019-01-14    kp1901.tab_2019-01-14
2019-01-15    kp1901.tab_2019-01-15
2019-01-16    kp1901.tab_2019-01-16
2019-01-17    kp1901.tab_2019-01-17
2019-01-18    kp1901.tab_2019-01-18
2019-01-19    kp1901.tab_2019-01-19
2019-01-20    kp1901.tab_2019-01-20
2019-01-21    kp1901.tab_2019-01-21
2019-01-22    kp1901.tab_2019-01-22
2019-01-23    kp1901.tab_2019-01-23
2019-01-24    kp1901.tab_2019-01-24
2019-01-25    kp1901.tab_2019-01-25
2019-01-26    kp1901.tab_2019-01-26
2019-01-27    kp1901.tab_2019-01-27
2019-01-28    kp1901.tab_2019-01-28
2019-01-29    kp1901.tab_2019-01-29
2019-01-30    kp1901.tab_2019-01-30
2019-01-31    kp1901.tab_2019-01-31
2019-02-01    kp1902.tab_2019-02-01
Freq: D, dtype: object
```
Everything works fine for January, but Feb drops the rest of the month.  Checking `kp.bounds` produces
```
([datetime.datetime(2019, 1, 1, 0, 0)], [datetime.datetime(2019, 2, 1, 0, 0)])
```
Repeating the whole process for TIMED/SEE, which has a similar monthly data file, produces
```
([datetime.datetime(2019, 1, 1, 0, 0)], [datetime.datetime(2019, 2, 28, 0, 0)])
```
as it should.

**Comment by jklenzing** [comment]
Not sure if the bounds is causing the file problem or vice versa.  Documenting both.

**Comment by rstoneback** [comment]
Thanks for catching this and documenting it. My initial guess is the way the routine fakes daily files from monthly. Kp was the first to do so.

On Mar 8, 2019, at 6:38 AM, Jeff Klenzing <notifications@github.com<mailto:notifications@github.com>> wrote:


Not sure if the bounds is causing the file problem or vice versa. Documenting both.

**Comment by jklenzing** [comment]
Found the root of the problem.  
```
pds_offset = pds.DateOffset(months=1) - pds.DateOffset(days=1)   
```
has no definition if defined alone, rather than operating directly on an index
```
out.index[-1] + pds.DateOffset(months=1) - pds.DateOffset(days=1)
```
Pretty sure this was one of my "simplifications" at some point.  My bad.  Fixing in #192.

**Comment by jklenzing** [comment]
Closed with merge of #192.
